portfolio.py

```diff
@@ -2,13 +2,40 @@
 Allocation of the asset.
 """
 
+import numpy as np
+from thousandaire.constants import TRADING_INSTRUMENTS
+
 class Portfolio(dict):
     """
     Set the portfolio
     """
-    def __init__(self, date):
+    def __init__(self, date, target=None, np_array=None):
         dict.__init__(self)
         self.date = date
+        if target is not None:
+            self.decode_nparray(target, np_array)
+
+    def decode_nparray(self, target, np_array):
+        """
+        Decode numpy array into portfolio.
+        """
+        if len(np_array) != len(TRADING_INSTRUMENTS[target]):
+            raise TypeError("Cannot decode into the target type.")
```

Comment:
`ValueError('Input dimension does not match number of instruments.')`
In numpy, dimension mismatch leads to ValueError. I think it's better to be consistent with it.